lib/types/keys.js

```diff
@@ -407,8 +405,8 @@ module.exports = Any._extend({
                     Assert(rename.from !== from, 'Cannot rename the same key multiple times');
                 }
 
-                if (to instanceof Template) {
-                    obj.$_mutateRegister(to);
+                if (Template.isTemplate(to)) {
+                    obj.$_mutateRegister(to._template);
```

Comment:
I believe `to._template` is an array of template parts (strings and objects representing dynamic parts), and I don't think this input is something `$_mutateRegister()` would expect to handle (the input is likely being ignored from what I can tell).  I think I see the underlying issue here, though.  In #9 I removed [some code](https://github.com/hapijs/validate/pull/9/files#diff-5207374d359764621ba8995d8ed346dbL335) which I expect was causing errors here the way it was originally written.  Templates just don't support refs in validate they way they do in joi— Eran removed it [here](https://github.com/hapijs/validate/commit/d6c04b8a61589932b8c94ee73ad8938d787d4875#diff-8dc0fb1119587e3ecd8cc6a9c9969d05L143).  So I think it might actually make sense to remove this code altogether and enforce that `to` must be a string.  What do you think?

Reply:
Now that #9 is merged, if you rebase this then that might help 👍

Reply:
Thanks, I agree that is likely the issue. I'll rebase and test.